### PR TITLE
Address compile warnings & trivial improvements

### PR DESCRIPTION
**Pull request by ricrogz**
This PR addresses most of the warnings that are issued during build, as well as some easy improvements in `GETAWAY.cpp`.

This patch exposed a failure in test `testGETAWAYcustom`, inside file `testRDFcustom.cpp`. In this test, function `RDKit::Descriptors::GETAWAY` was being called with argument `precision = 0.0001`. As the `precision` parameter was declared as `int`, it was being casted into a 0, causing the values in the function to be reduced to 0 decimals precision. This wrong casting was being flagged as a warning by the compiler, and as soon as I changed the precision to 4, meaning four decimals precision, as in 0.0001, the test failed. It only passes for `precision = 0` or `precision = 1`.

The changes in `GETAWAY.cpp` and updating `abs()` called on real values to `fabs()` were done while I tried to spot the problem with the test, which in the end I have not been able to find.

### CODE REVIEW

File: Code/ForceField/MMFF/Params.cpp [hunk 1]
```diff
@@ -2157,14 +2157,14 @@ MMFFCovRadPauEleCollection *MMFFCovRadPauEleCollection::getMMFFCovRadPauEle(
...
-  if (mmffCovRadPauEle == "") {
+  if (mmffCovRadPauEle.empty()) {
     mmffCovRadPauEle = defaultMMFFCovRadPauEle;
   }
   std::istringstream inStream(mmffCovRadPauEle);
   std::string inLine = RDKit::getLine(inStream);
   while (!(inStream.eof())) {
     if (inLine[0] != '*') {
-      MMFFCovRadPauEle mmffCovRadPauEleObj;
+      MMFFCovRadPauEle mmffCovRadPauEleObj{};
```
Comment (greglandrum) [inline Code/ForceField/MMFF/Params.cpp]: It's not clear to me why this construct is an improvement over what was there before.

File: Code/GraphMol/Depictor/RDDepictor.cpp [hunk 2]
```diff
@@ -418,7 +418,7 @@ unsigned int compute2DCoordsMimicDistMat(
     RDKit::ROMol &mol, const DOUBLE_SMART_PTR *dmat, bool canonOrient,
     bool clearConfs, double weightDistMat, unsigned int nFlipsPerSample,
     unsigned int nSamples, int sampleSeed, bool permuteDeg4Nodes,
-    bool forceRDKit) {
+    bool /* forceRDKit */) {
```
Comment (greglandrum) [inline Code/GraphMol/Depictor/RDDepictor.cpp]: We normally suppress this warning with `RDUSED_PARAM(forceRDKit)`

File: Code/ForceField/MMFF/Params.cpp (hunk 1 above)
Comment (ricrogz) [reply Code/ForceField/MMFF/Params.cpp]: clang-tidy suggested those. I didn't mean to include them, they must have slipped in.

File: Code/GraphMol/Depictor/RDDepictor.cpp (hunk 2 above)
Comment (ricrogz) [reply Code/GraphMol/Depictor/RDDepictor.cpp]: Ok, updating those.

**Comment by greglandrum** [comment]
@thegodone the typo fix in this PR has unconvered a failing test in the custom RDF tests. I guess this is because you generated the reference data using the same bug. Would you mind taking a look at this please?

**Comment by ricrogz** [comment]
@greglandrum, done! I restored the 0.001 in the call to RDKit::Descriptors::GETAWAY, and marked it with a comment.

**Comment by greglandrum** [comment]
Thanks for this @ricrogz

**Comment by thegodone** [comment]
I will check now

Envoyé de mon iPhone

**Comment by thegodone** [comment]
that's right. I didn't see this message before.

I can generate a new dataset now,

BR

Guillaume

Le lun. 8 oct. 2018 à 18:37, Greg Landrum <notifications@github.com> a
écrit :

**Comment by ricrogz** [comment]
> Thanks for this @ricrogz

My pleasure!

**Pull request merged**